test/extensions/filters/network/thrift_proxy/mocks.h

```diff
@@ -258,7 +258,35 @@ class MockDecoderFilterCallbacks : public DecoderFilterCallbacks {
   std::shared_ptr<Router::MockRoute> route_;
 };
 
-class MockFilterConfigFactory : public ThriftFilters::FactoryBase<ProtobufWkt::Struct> {
+template <class ConfigProto> class MockFactoryBase : public NamedThriftFilterConfigFactory {
+public:
+  FilterFactoryCb
+  createFilterFactoryFromProto(const Protobuf::Message& proto_config,
+                               const std::string& stats_prefix,
+                               Server::Configuration::FactoryContext& context) override {
+    const auto& typed_config = dynamic_cast<const ConfigProto&>(proto_config);
+    return createFilterFactoryFromProtoTyped(typed_config, stats_prefix, context);
```

Comment:
I think this is only used in one spot, so I think we could skip the template and make this `class MockFilterConfigFactory : public NamedThriftFilterConfigFactory` and merge the implementation of `createFilterFactoryFromProtoTyped` below into this method.

Reply:
thanks for the pointer, changes should be updated